File: example/java/pact_validation_with_oas/one_consumer_one_provider/consumer/src/test/java/br/com/zup/pact/consumer/pact/AccountPactTest.java
package br.com.zup.pact.consumer.pact;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;

import au.com.dius.pact.consumer.MockServer;
import au.com.dius.pact.consumer.dsl.*;
import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
import au.com.dius.pact.consumer.junit5.PactTestFor;
import au.com.dius.pact.core.model.RequestResponsePact;
import au.com.dius.pact.core.model.annotations.Pact;
import br.com.zup.pact.consumer.dto.BalanceDTO;
import com.google.gson.Gson;
import java.io.IOException;
import java.math.BigDecimal;
import java.util.HashMap;
import java.util.Map;
import org.apache.commons.collections4.MapUtils;
import org.apache.commons.io.IOUtils;
import org.apache.http.HttpResponse;
import org.apache.http.client.fluent.Request;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpMethod;
import org.springframework.http.MediaType;

@ExtendWith(PactConsumerTestExt.class)
@PactTestFor(providerName = "provider")
public class AccountPactTest {

    private static final String BALANCE_URL_WORKING = "/v1/accounts/1/balance/";
    private static final String BALANCE_URL_NOT_WORKING = "/v1/accounts/1000/balance/";
    private static final String CONSUMER_NAME = "consumer2";


    private Map<String, String> headers = MapUtils.putAll(new HashMap<>(), new String[] {
            HttpHeaders.CONTENT_TYPE,
            MediaType.APPLICATION_JSON_VALUE
    });

    private Gson gson = new Gson();

    @Pact(consumer = CONSUMER_NAME)
    public RequestResponsePact balanceEndpointTest(PactDslWithProvider builder) {
        PactDslJsonBody bodyResponse = new PactDslJsonBody()
                .integerType("accountId")
                .numberType("balance")
                .stringValue("accountType", "PERSONAL");

        return builder
                .given("get balance of accountId 1")
                .uponReceiving("A request to " + BALANCE_URL_WORKING)
                .path(BALANCE_URL_WORKING)
                .method(HttpMethod.GET.name())
                .willRespondWith()
                .headers(headers)
                .status(200)
                .body(bodyResponse)
                .toPact();
    }

    @Pact(consumer = CONSUMER_NAME)
    public RequestResponsePact balanceEndpointNotWorkingTest(PactDslWithProvider builder) {
        return builder
                .given("No accounts exist from accountId 1000")
                .uponReceiving("A request to " + BALANCE_URL_NOT_WORKING)
                .path(BALANCE_URL_NOT_WORKING)
                .method(HttpMethod.GET.name())
                .willRespondWith()
                .status(404)
                .toPact();
    }

    @Test
    @PactTestFor(pactMethod = "balanceEndpointTest")
    void testBalanceWorking(MockServer mockServer) throws IOException {
        HttpResponse httpResponse = Request.Get(mockServer.getUrl() + BALANCE_URL_WORKING).execute().returnResponse();
        assertThat(httpResponse.getStatusLine().getStatusCode(), is(equalTo(200)));
    }

    @Test
    @PactTestFor(pactMethod = "balanceEndpointNotWorkingTest")
    void testBalanceNotWorking(MockServer mockServer) throws IOException {
        HttpResponse httpResponse = Request.Get(mockServer.getUrl() + BALANCE_URL_NOT_WORKING).execute().returnResponse();
        assertThat(httpResponse.getStatusLine().getStatusCode(), is(equalTo(404)));
    }

}

File: examples/kotlin/one-consumer-one-provider/consumer/src/test/kotlin/br/com/zup/pact/consumer/pact/AccountPactTest.kt
package br.com.zup.pact.consumer.pact

import au.com.dius.pact.consumer.MockServer
import au.com.dius.pact.consumer.dsl.PactDslJsonBody
import au.com.dius.pact.consumer.dsl.PactDslWithProvider
import au.com.dius.pact.consumer.junit5.PactConsumerTestExt
import au.com.dius.pact.consumer.junit5.PactTestFor
import au.com.dius.pact.core.model.RequestResponsePact
import au.com.dius.pact.core.model.annotations.Pact
import br.com.zup.pact.consumer.constants.CONSUMER_NAME
import br.com.zup.pact.consumer.constants.PROVIDER_NAME
import br.com.zup.pact.consumer.constants.PORT
import org.apache.http.HttpResponse
import org.apache.http.client.fluent.Request
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(PactConsumerTestExt::class)
@PactTestFor(providerName = PROVIDER_NAME, port = PORT)
class AccountPactTest {

    private val balanceUrlWorking: String = "/v1/accounts/1/balance"
    private val balanceUrlNotWorking: String = "/v1/accounts/1000/balance"

    private val header = mapOf("Content-Type" to "application/json")

    @Pact(consumer = CONSUMER_NAME)
    fun balanceEndpointTest(builder: PactDslWithProvider): RequestResponsePact {
        val bodyResponse: PactDslJsonBody = PactDslJsonBody()
                .integerType("accountId")
                .integerType("clientId")
                .numberType("balance")

        return builder.given("get balance of accountId 1")
                .uponReceiving("A request to $balanceUrlWorking")
                .path(balanceUrlWorking)
                .method("GET")
                .willRespondWith()
                .headers(header)
                .status(200)
                .body(bodyResponse)
                .toPact()
    }

    @Pact(consumer = CONSUMER_NAME)
    fun balanceEndpointNotWorkingTest(builder: PactDslWithProvider): RequestResponsePact {
        return builder.given("No accounts exist from accountId 1000")
                .uponReceiving("A request to $balanceUrlNotWorking")
                .path(balanceUrlNotWorking)
                .method("GET")
                .willRespondWith()
                .status(404)
                .toPact()
    }

    @Test
    @PactTestFor(pactMethod = "balanceEndpointTest")
    fun testBalanceWorking(mockServer: MockServer) {
        val httpResponse: HttpResponse = Request.Get(mockServer.getUrl() + balanceUrlWorking)
                .execute().returnResponse()
        Assertions.assertThat(httpResponse.statusLine.statusCode).isEqualTo(200)
    }

    @Test
    @PactTestFor(pactMethod = "balanceEndpointNotWorkingTest")
    fun testBalanceNotWorking(mockServer: MockServer) {
        val httpResponse: HttpResponse = Request.Get(mockServer.getUrl() + balanceUrlNotWorking)
                .execute().returnResponse()
        Assertions.assertThat(httpResponse.statusLine.statusCode).isEqualTo(404)
    }

}